test/integration/options/posixExitCodes.spec.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+var helpers = require('../helpers');
+var runMocha = helpers.runMocha;
+
+describe('--posix-exit-codes', function () {
+  describe('when enabled, and with node options', function () {
+    var args = ['--no-warnings', '--posix-exit-codes'];
+
+    it('should exit with code 134 on SIGABRT', function (done) {
+      var fixture = 'signals-sigabrt.fixture.js';
+      runMocha(fixture, args, function postmortem(err, res) {
+        if (err) {
+          return done(err);
+        }
+        expect(res.code, 'to be', 134);
+        done();
+      });
+    });
+
+    it('should exit with code 143 on SIGTERM', function (done) {
+      var fixture = 'signals-sigterm.fixture.js';
+      runMocha(fixture, args, function postmortem(err, res) {
+        if (err) {
+          return done(err);
+        }
+        expect(res.code, 'to be', 143);
+        done();
+      });
+    });
+
+    it('should exit with code 0 even if there are test failures', function (done) {
+      var fixture = 'failing.fixture.js';
+      runMocha(fixture, args, function postmortem(err, res) {
+        if (err) {
+          return done(err);
+        }
+        expect(res.code, 'to be', 0);
+        done();
+      });
+    });
+  });
+
```

Comment:
Don't understand why L44-55 is desired -- shouldn't it just do like it (unhelpfully) did before?

It doesn't seem helpful to get a NULL return code from a signal.

Reply:
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/signal.h.html

Use table above to ensure all the signals being handled are as expected.

Reply:
> Don't understand why L44-55 is desired -- shouldn't it just do like it (unhelpfully) did before?
> 
> It doesn't seem helpful to get a NULL return code from a signal.

This is to assert that the existing behavior doesn't change when the new option isn't given and ensure there's no unexpected breaking change.

Reply:
> https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/signal.h.html
> 
> Use table above to ensure all the signals being handled are as expected.

In this clause of the suite, the signal shouldn't matter since we're testing the behavior _without_ the `--posix-exit-codes` flag and want to ensure that mocha continues to exit with the number of failed tests. In the previous clause we test scenarios where `--posix-exit-codes` is present, but there isn't really any special handling to be done beyond ensuring that the appropriate `exit_code` is set. We have test cases there to cover a couple of common signals... adding a test for every signal in `os.constants.signals` would be overkill imho and not adding any real value since they're all simply passed on to the exit call with a +128 offset.

Reply:
PS the null code is just the existing mocha behavior, we didn't change that her; we're just asserting that the old behavior keeps working the same way.

Reply:
> > https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/signal.h.html
> > Use table above to ensure all the signals being handled are as expected.
> 
> [SNIP] We have test cases there to cover a couple of common signals... adding a test for every signal in `os.constants.signals` would be overkill imho and not adding any real value since they're all simply passed on to the exit call with a +128 offset.

Fine. I didn't see where you were setting signal handlers. The table was to ensure you only did so for signals with either a T or A action (ignoring SIGKILL obviously).

> PS the null code is just the existing mocha behavior, we didn't change that her; we're just asserting that the old behavior keeps working the same way.

Perhaps that null code aspect was added when Mocha went parallel then. But all this comes back to the shell expecting an integer return -- not sure there's such a thing as a null code -- not being converted to zero (no failure), is it?

Reply:
`process.exit(null)` results in a `0` exit code in the shell, yes.

Reply:
Are you saying the existing behavior is to terminate with successful return code (0) when a signal handler is (abnormally) invoked?

Reply:
Yes, exactly :-) ... that existing behavior is the root cause of the issues in the PR description, hence the need for an option to tell mocha _not_ to do this, and use a standard posix / shell exit code instead: "This helps to solve issues for toolchains that expect standard posix [/ unix shell] exit codes, for example by preventing out-of-memory crashes from being silently ignored."

Reply:
PS. Even though the GNU libc manual warns against using exit codes to indicate the number of failed tests ("Don’t try to use the number of errors as the exit status... it does not work"), this PR isn't meant to express an opinion on whether or not mocha's current behavior is good.  From the linked issues, it seems that some tools now depend on mocha's existing behavior, and to avoid breaking those, this command line option should preserve the original behavior while providing a solution for toolchains that expect standard posix / shell exit codes.

Reply:
Preserving so many different broken behaviors will be its own maintenance nightmare. My original was only to separate Mocha #errors from its exit code, while providing a sane alternative. I certainly understand sometimes  things need be done piecemeal to work them into the codebase, but IMHO deprecation warnings should be given for compatible-but-[borked](https://cheezburger.com/3878624512/borked) stuff like that (and the entire nonsense be removed upon the next major).

Reply:
I agree and I think what you suggested will be a reasonable path forward as a next step if the maintainers want to make this the default behavior. For this PR, changing the default behavior would be out of scope. Any remaining concerns on this code section in particular?

Reply:
I can go ahead and remove this test case - in a way it's ensconcing a behavior that probably should never have existed and would make it just a bit harder to fix that behavior going forward.